test/omitempty/combos/marshaler/omitempty.pb.go

```diff
@@ -0,0 +1,917 @@
+		0xd9, 0x7f, 0x0b, 0xbf, 0xda, 0xf5, 0x05, 0xf3, 0x27, 0xbb, 0x14, 0x5e, 0xf8, 0xf9, 0x14, 0xe4,
+		0x9b, 0x3d, 0xdd, 0xab, 0x93, 0x59, 0x7e, 0x02, 0x26, 0x1b, 0xa6, 0x89, 0x9c, 0x2d, 0xb6, 0x55,
+		0xc9, 0xd7, 0xcd, 0xc2, 0xf9, 0xf9, 0xa5, 0xc1, 0x0b, 0x7d, 0xe5, 0x25, 0xa2, 0x29, 0x87, 0x01,
+		0xd2, 0x06, 0x88, 0x54, 0x60, 0x79, 0x3e, 0x49, 0x2a, 0x8e, 0x64, 0x25, 0x73, 0xfb, 0x1f, 0xcf,
+		0x8c, 0xc9, 0x43, 0x48, 0x69, 0x03, 0x4a, 0x44, 0xe6, 0xeb, 0xb3, 0x5f, 0xed, 0x1d, 0xc7, 0x95,
+		0xc3, 0x5c, 0xaf, 0xff, 0xd3, 0x19, 0x41, 0x8e, 0x60, 0xe7, 0xcf, 0x40, 0x96, 0x48, 0xa4, 0x39,
+		0x48, 0x5f, 0xb1, 0x2c, 0x32, 0xb9, 0x34, 0x7b, 0x37, 0x16, 0xac, 0x6c, 0xdc, 0xbe, 0x53, 0x19,
+		0xfb, 0xde, 0x9d, 0xca, 0xd8, 0xdf, 0xdf, 0xa9, 0x8c, 0xbd, 0x71, 0xa7, 0x22, 0xbc, 0x75, 0xa7,
+		0x22, 0xbc, 0x73, 0xa7, 0x22, 0xbc, 0x7b, 0xa7, 0x22, 0xdc, 0x3a, 0xac, 0x08, 0xaf, 0x1f, 0x56,
+		0x84, 0xaf, 0x1e, 0x56, 0x84, 0x6f, 0x1e, 0x56, 0x84, 0xef, 0x1c, 0x56, 0x84, 0xdb, 0x87, 0x15,
+		0xe1, 0x7b, 0x87, 0x95, 0xb1, 0x37, 0x0e, 0x2b, 0xc2, 0x5b, 0x87, 0x95, 0xb1, 0x77, 0x0e, 0x2b,
+		0xc2, 0xbb, 0x87, 0x95, 0xb1, 0x5b, 0x3f, 0xa8, 0x8c, 0xfd, 0x7f, 0x00, 0x00, 0x00, 0xff, 0xff,
+		0xf9, 0x99, 0x4a, 0x7a, 0xed, 0x34, 0x00, 0x00,
+	}
+	r := bytes.NewReader(gzipped)
+	gzipr, err := compress_gzip.NewReader(r)
+	if err != nil {
+		panic(err)
+	}
+	ungzipped, err := io_ioutil.ReadAll(gzipr)
+	if err != nil {
+		panic(err)
+	}
+	if err := github_com_gogo_protobuf_proto.Unmarshal(ungzipped, d); err != nil {
+		panic(err)
+	}
+	return d
+}
+func (this *OmitEmpty) VerboseEqual(that interface{}) error {
+	if that == nil {
+		if this == nil {
+			return nil
+		}
+		return fmt.Errorf("that == nil && this != nil")
+	}
+
+	that1, ok := that.(*OmitEmpty)
+	if !ok {
+		that2, ok := that.(OmitEmpty)
+		if ok {
+			that1 = &that2
+		} else {
+			return fmt.Errorf("that is not of type *OmitEmpty")
+		}
+	}
+	if that1 == nil {
+		if this == nil {
+			return nil
+		}
+		return fmt.Errorf("that is type *OmitEmpty but is nil && this != nil")
+	} else if this == nil {
+		return fmt.Errorf("that is type *OmitEmpty but is not nil && this == nil")
+	}
+	if !this.InnerNullable.Equal(that1.InnerNullable) {
+		return fmt.Errorf("InnerNullable this(%v) Not Equal that(%v)", this.InnerNullable, that1.InnerNullable)
+	}
+	if !this.InnerNotNullable.Equal(&that1.InnerNotNullable) {
+		return fmt.Errorf("InnerNotNullable this(%v) Not Equal that(%v)", this.InnerNotNullable, that1.InnerNotNullable)
+	}
+	if !this.InnerOmitEmpty.Equal(&that1.InnerOmitEmpty) {
+		return fmt.Errorf("InnerOmitEmpty this(%v) Not Equal that(%v)", this.InnerOmitEmpty, that1.InnerOmitEmpty)
+	}
+	if !bytes.Equal(this.XXX_unrecognized, that1.XXX_unrecognized) {
+		return fmt.Errorf("XXX_unrecognized this(%v) Not Equal that(%v)", this.XXX_unrecognized, that1.XXX_unrecognized)
+	}
+	return nil
+}
+func (this *OmitEmpty) Equal(that interface{}) bool {
```

Comment:
Should we follow the pattern set by `Equal` of exposing a `gogoproto.empty` option to generate an `IsEmpty`/`Empty` method?

Reply:
I'm not supercomfortable with working in this repo, I'd rather not complicate things here. It will be trivial to write it manually wherever we want to use this.